libsolutil/JSON.cpp

```diff
@@ -115,18 +115,34 @@ Json::Value removeNullMembers(Json::Value _json)
 
 string jsonPrettyPrint(Json::Value const& _input)
 {
-	static map<string, Json::Value> settings{{"indentation", "  "}, {"enableYAMLCompatibility", true}};
-	static StreamWriterBuilder writerBuilder(settings);
-	string result = print(_input, writerBuilder);
-	boost::replace_all(result, " \n", "\n");
-	return result;
+	return jsonPrint(_input, JsonPrintingFormat{ JsonPrintingFormat::Format::Pretty});
 }
 
 string jsonCompactPrint(Json::Value const& _input)
 {
-	static map<string, Json::Value> settings{{"indentation", ""}};
-	static StreamWriterBuilder writerBuilder(settings);
-	return print(_input, writerBuilder);
+	return jsonPrint(_input, JsonPrintingFormat{ JsonPrintingFormat::Format::Compact });
+}
+
+/// Serialise the JSON object (@a _input) using specified format (@a _format)
+string jsonPrint(Json::Value const& _input, JsonPrintingFormat const& _format)
+{
+	map<string, Json::Value> settings;
+	if (_format.format == JsonPrintingFormat::Format::Pretty)
+	{
+		settings["indentation"]= string(_format.indent, ' ');
```

Comment:
```suggestion
		settings["indentation"] = string(_format.indent, ' ');
```
There's also one below.